nx/lib/nx/lin_alg.ex

```diff
@@ -2138,6 +2138,105 @@ defmodule Nx.LinAlg do
     Nx.sum(s > tol)
   end
 
+  @doc """
+  Return the least-squares solution to a linear matrix equation Ax = b.
+
+  ## Examples
+
+      iex> Nx.LinAlg.lstsq(Nx.tensor([[1, 2], [2, 3]]), Nx.tensor([1, 2]))
```

Comment:
How about we rename it to least_squares?

Reply:
I agree with main contoributor's intention to change the name to `least_squares`, since I was just imitating `numpy.linalg.lstsq`. I'll change it to `least_squares`.

Reply:
In general, we try to use the longer name for readability. pinv is one of the few exceptions to this